pyfar/classes/audio.py

```diff
@@ -1555,15 +1565,17 @@ def _assert_match_for_arithmetic(data: tuple, domain: str, division: bool,
             if np.asarray(d).dtype.kind not in ["i", "f", "c"]:
                 raise ValueError(
                     "Input must be of type Signal, int, float, or complex")
-            if np.asarray(d).dtype.kind == "c" and domain == 'time':
-                raise ValueError(
-                    "Complex input can not be applied in the time domain.")
+            if audio_type == (Signal or TimeData):
+                # if the first data was a Signal or TimeData set
+                # complex flag if the non signal input is complex-valued
+                if np.asarray(d).dtype.kind == "c":
+                    complex = True
```

Comment:
I think there are two things that might break here:

- This is in the loop over the input argument `data`, hence `audio_type` can not be used here. In case the first element of `data` is an array, you would enter this else statement and and `auio_type` would be `None`, but it might be changed later if the second element is a Signal.
- If the operation is in the frequency domain, the array data could be complex without the need to cast all other Signals contained in `data` to `complex = True`.

The simplest solution that comes to my mind is

```
if domain == 'time' and np.asarray(d).dtype.kind == "c":
    complex = True
```

Would this work? In any case we should have test cases for that:

- time domain operation (4 tests)
  - real-valued array as first (second) element in `data`
  - complex-valued array as first (second) element in `data`
- frequency domain operation (same 4 tests as above)